rest/controller.go

```diff
@@ -0,0 +1,213 @@
+package rest
+
+import (
+	"net/http"
+	"net/http/pprof"
+	"runtime"
+
+	"github.com/labstack/echo/v4"
+	"github.com/proemergotech/log/v3"
+	"github.com/proemergotech/log/v3/echolog"
+
+	"github.com/artofimagination/mysql-resources-db-go-service/models"
+	httpModels "github.com/artofimagination/mysql-resources-db-go-service/models/http"
+	"github.com/artofimagination/mysql-resources-db-go-service/service"
+)
+
+type controller struct {
+	echoEngine *echo.Echo
+	svc        *service.Service
+	debugPProf bool
+}
+
+func NewController(
+	echoEngine *echo.Echo,
+	svc *service.Service,
+	debugPProf bool,
+) Controller {
+	return &controller{
+		echoEngine: echoEngine,
+		svc:        svc,
+		debugPProf: debugPProf,
+	}
+}
+
+func (c *controller) Start() {
+	if c.debugPProf {
+		runtime.SetBlockProfileRate(1)
+		runtime.SetMutexProfileFraction(5)
+
+		c.echoEngine.GET("/debug/pprof/*", echo.WrapHandler(http.HandlerFunc(pprof.Index)))
+		c.echoEngine.GET("/debug/pprof/cmdline", echo.WrapHandler(http.HandlerFunc(pprof.Cmdline)))
+		c.echoEngine.GET("/debug/pprof/profile", echo.WrapHandler(http.HandlerFunc(pprof.Profile)))
+		c.echoEngine.GET("/debug/pprof/symbol", echo.WrapHandler(http.HandlerFunc(pprof.Symbol)))
+		c.echoEngine.GET("/debug/pprof/trace", echo.WrapHandler(http.HandlerFunc(pprof.Trace)))
+	}
+
+	c.echoEngine.GET("/", func(eCtx echo.Context) error {
+		return eCtx.String(http.StatusOK, "Hi! I am a user database server!")
+	})
+
+	c.echoEngine.GET("/healthcheck", func(eCtx echo.Context) error {
+		return eCtx.String(http.StatusOK, "ok")
+	})
+
+	c.echoEngine.POST("/add-resource", func(eCtx echo.Context) error {
+		resource := &models.Resource{}
+		if err := eCtx.Bind(resource); err != nil {
```

Comment:
Is it binding the parameters or json map of incoming request to the resource?
Yes. If you dive into the code of this method, you will see this could bind json, query, url_params to your destination struct. Also handles query param list (e.g. ids) and other stuff.